lib/SILOptimizer/Mandatory/TFConstExpr.cpp

```diff
@@ -1202,6 +1207,10 @@ ConstExprFunctionState::getSingleWriterAddressValue(SILValue addr) {
       if (memoryObject->getValue().getKind() != SymbolicValue::UninitMemory)
         return evaluator.getUnknown(addr, UnknownReason::Default);
 
+      // Set `addr` to the address of the memory we want to initialize, so that
+      // the callee can find and initialize it.
+      setValue(addr, memoryAddress);
```

Comment:
do we need to do the same in the "if (auto *si = dyn_cast<StoreInst>(user))" branch?

Reply:
No, the others set the `memoryObject` directly without looking it up through `addr`, so we only need to `setValue(addr, memoryAddress)` at the very end when they are completely successful (which happens on L1302).

Reply:
Update: I'm running into the same bug from a different context, and have some suggestions. Let's have a quick chat today?